examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -43,6 +52,44 @@ class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
         this._priority.on("valueChanged", () => {
             this.emit("priorityChanged");
         });
+        this.diffType = "none";
+        this.diffName = "";
+        this.diffPriority = Number.POSITIVE_INFINITY;
+    }
+    public externalNameChanged = async (savedName: string): Promise<void> => {
+        this.diffName = savedName;
+        this.diffType = "change";
+        this.emit('externalNameChanged');
+    }
+    public externalPriorityChanged = async (savedPriority: number): Promise<void> => {
+        this.diffPriority = savedPriority;
+        this.diffType = "change";
+        this.emit('externalPriorityChanged');
+    }
+    public acceptChange = async (): Promise<void> => {
+        if (this.diffPriority !== this.DEFAULT_PRIORITY) {
+            this._priority.set(this.diffPriority);
+        }
+        if (this.diffName !== '') {
+            const oldString = this._name.getText()
+            console.log(this.name.getText());
+            this._name.replaceText(0, oldString.length, this.diffName);
+            console.log(this.name.getText());
+        }
+        this.diffPriority = this.DEFAULT_PRIORITY;
+        this.diffType = "none";
+        this.emit('externalPriorityChanged');
+        this.diffName = '';
+        this.diffType = "none";
+        this.emit('externalNameChanged');
```

Comment:
Should we only emit the events when the specific fields changed? I.e. inside the ifs above.

Reply:
These emits will trigger removing the UI from the screen.

Reply:
OH hm I misread that the first time around. I see what you're saying. Let me see.